### DISCUSSION TITLE
ModbusTCP without ServerID

### DISCUSSION THREAD

**Discussion by Typhoor** [eModbus/eModbus#303] [Q&A] [answered]
I have an application with a ModbusServerTCPAsync where the querying system will use different UnitIDs depending on the situation. 
The "Modbus Standard" does not enforce a ServerID for ModbusTCP, but rather speaks about a UnitID which should be just returned or used for routing (like in bridge situations).
![grafik](https://github.com/eModbus/eModbus/assets/18496439/85aa8359-8750-4054-9ed5-a51e2600663d)
 
Registering Worker on ModbusServerTCPAsync always asks for a ServerID ?!? 
Shouldn't it be possible use use ServerID=0 as BroadcastID to answer all UnitIDs.

So how shall I handle this, other then registering 1...247 Worker or change the libraries source code ?
Any ideas ? Hidden possibilities?

**Comment by Miq1** [comment, marked as answer]
First, broadcast is not suitable for your task, because the standard will not allow any responses for a broadcast request.
The setup you are describing of course may only work on Modbus TCP, since the server id is not used for addressing normally.
So what you will need is some "promiscuous" TCP server.
The library in fact has such in the form of a listener, that will get all messages regardless, but is not able to respond.
A solution could be a kind of worker being able to register for ANY_SERVER (and a function code or potentially ANY_FUNCTION as well). While this would be not too hard to implement, it will require some deeper consideration of possible side effects; f.i. it could break the broadcast and listener code.
I have not much free capacity to do that on the fly, but would be willing to look after it in the next few weeks. 
If that time frame is acceptable for you I will create a feature request from this discussion.

**Reply by Typhoor** [reply]
Thanx.
I will change that in my local lib and test it.
And you can decide whether to take that up to the official side.
